consumer/consumer.go

```diff
@@ -470,7 +470,7 @@ func (dc *defaultConsumer) lock(mq *primitive.MessageQueue) bool {
 			pq.lastConsumeTime = time.Now()
 			pq.lastLockTime = time.Now()
 		}
-		if _mq.Equals(mq) {
+		if _mq == *mq {
```

Comment:
Could you provide some explanation for understanding?

Reply:
@wenfengwang  equals method impl is incorrect. u can see that as bellow: 
```
func (mq MessageQueue) Equals(queue *MessageQueue) bool {
	// TODO
	return mq.BrokerName == queue.BrokerName && mq.Topic == queue.Topic && mq.QueueId == mq.QueueId
}
```
the last equal is `mq.QueueId == mq.QueueId`, which is wrong, and with golang, we can use '==' to check equal when field in obj is plain